src/Controllers/API/UpdateStudentProfileController.php

```diff
@@ -0,0 +1,182 @@
+<?php
+
+namespace Portal\Controllers\API;
+
+use Exception;
+use Portal\Abstracts\Controller;
+use Portal\Interfaces\ApplicantModelInterface;
+use Portal\Models\ApplicantModel;
+use Portal\Sanitisers\ApplicantSanitiser;
+use Portal\Validators\ApplicantValidator;
+use Psr\Http\Message\ResponseInterface as Response;
+use Psr\Http\Message\ServerRequestInterface as Request;
+
+class UpdateStudentProfileController extends Controller
+{
+    private ApplicantModel $applicantModel;
+
+    public function __construct(ApplicantModel $applicantModel)
+    {
+        $this->applicantModel = $applicantModel;
+    }
+
+    public function __invoke(Request $request, Response $response, array $args): Response
+    {
+        $updatedStudentProfileData = $request->getParsedBody();
+
+        if (
+            !empty($_SESSION['studentLogin']) &&
+            $_SESSION['studentLogin'] &&
+            $_SESSION['studentId'] == $updatedStudentProfileData['id'] ||
+            !empty($_SESSION['loggedIn']) &&
+            $_SESSION['loggedIn']
```

Comment:
it's not entirely clear here which operator takes precedence because you have one || among lots of &&s, so it would help to put some brackets to make it clear which conditions go together